ui/component/txoList/view.jsx

```diff
@@ -0,0 +1,264 @@
+    }
+  }
+
+  if (currentUrlParams.page) params[TXO.PAGE] = Number(page);
+  if (currentUrlParams.pageSize) params[TXO.PAGE_SIZE] = Number(pageSize);
+
+  function handleChange(delta: Delta) {
+    const url = updateUrl(delta);
+    history.push(url);
+  }
+
+  function updateUrl(delta: Delta) {
+    const newUrlParams = new URLSearchParams();
+    switch (delta.dkey) {
+      case TXO.PAGE:
+        if (currentUrlParams.type) {
+          newUrlParams.set(TXO.TYPE, currentUrlParams.type);
+        }
+        if (currentUrlParams.subtype) {
+          newUrlParams.set(TXO.SUB_TYPE, currentUrlParams.subtype);
+        }
+        if (currentUrlParams.active) {
+          newUrlParams.set(TXO.ACTIVE, currentUrlParams.active);
+        }
+        newUrlParams.set(TXO.PAGE, delta.value);
+        break;
+      case TXO.TYPE:
+        newUrlParams.set(TXO.TYPE, delta.value);
+        if (delta.value === TXO.SENT || delta.value === TXO.RECEIVED) {
+          if (currentUrlParams.subtype) {
+            newUrlParams.set(TXO.SUB_TYPE, currentUrlParams.subtype);
+          } else {
+            newUrlParams.set(TXO.SUB_TYPE, 'all');
+          }
+        }
+        if (currentUrlParams.active) {
+          newUrlParams.set(TXO.ACTIVE, currentUrlParams.active);
+        }
+        newUrlParams.set(TXO.PAGE, String(1));
+        newUrlParams.set(TXO.PAGE_SIZE, currentUrlParams.pageSize);
+        break;
+      case TXO.SUB_TYPE:
+        if (currentUrlParams.type) {
+          newUrlParams.set(TXO.TYPE, currentUrlParams.type);
+        }
+        if (currentUrlParams.active) {
+          newUrlParams.set(TXO.ACTIVE, currentUrlParams.active);
+        }
+        newUrlParams.set(TXO.SUB_TYPE, delta.value);
+        newUrlParams.set(TXO.PAGE, String(1));
+        newUrlParams.set(TXO.PAGE_SIZE, currentUrlParams.pageSize);
+        break;
+      case TXO.ACTIVE:
+        if (currentUrlParams.type) {
+          newUrlParams.set(TXO.TYPE, currentUrlParams.type);
+        }
+        if (currentUrlParams.subtype) {
+          newUrlParams.set(TXO.SUB_TYPE, currentUrlParams.subtype);
+        }
+        newUrlParams.set(TXO.ACTIVE, delta.value);
+        newUrlParams.set(TXO.PAGE, String(1));
+        newUrlParams.set(TXO.PAGE_SIZE, currentUrlParams.pageSize);
+        break;
+    }
+    return `?${newUrlParams.toString()}`;
+  }
+
+  const paramsString = JSON.stringify(params);
+
+  useEffect(() => {
+    if (paramsString && fetchTransactionPage) {
+      const params = JSON.parse(paramsString);
+      fetchTransactionPage(params);
+    }
+  }, [paramsString, fetchTransactionPage]);
+
+  return (
+    <Card
+      title={__(`Transactions`)}
+      titleActions={
+        <div className="card__actions--inline">
+          <Button button="secondary" label={__('Refresh')} onClick={() => fetchTransactionPage(params)} />
+          {/* @if TARGET='app' */}
+          <Button button="primary" navigate={`/$/${PAGES.TRANSACTIONS}`} label={__('Full History')} />
```

Comment:
why is this app only?

Reply:
Relies on full transaction list that blows up sdk memory.

Reply:
There was a conversation about the export feature and the lack of fees info from txo_list that resulted in trying to save that component, at least for desktop app.